api/gateway.go

```diff
@@ -41,6 +45,10 @@ func (srv *Server) gatewayAddHandler(w http.ResponseWriter, req *http.Request, p
 // gatewayRemoveHandler handles the API call to remove a peer from the gateway.
 func (srv *Server) gatewayRemoveHandler(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
 	addr := modules.NetAddress(ps.ByName("netaddress"))
+	if err := addr.IsValid(); err != nil {
+		writeErorr(w, "invalid address: "+err.Error(), http.StatusBadRequest)
+		return
+	}
```

Comment:
>Regardless, I don't think validation should be performed in the API package. The API package strives to be a lightweight wrapper around the module objects. It should introduce as little additional logic as possible.

What about validation for host announce with specified address? Should we validate the address in `host.AnnounceAddress`?